### PR TITLE
Add CI for Linting and Tests

### PR DESCRIPTION
**Pull request by adason**
Use Github Action for Linting and Tests

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+          path: .venv
+          key: venv-${{ steps.full-python-version.outputs.version }}-${{ hashFiles('**/poetry.lock') }}
+      - name: Ensure cache is healthy
+        if: steps.cache.outputs.cache-hit == 'true'
+        shell: bash
+        run: poetry run pip --version >/dev/null 2>&1 || rm -rf .venv
+      - name: Install dependencies
+        run: poetry install
+        shell: bash
+      - name: Run pytest
```
Comment (Saurav-D) [inline .github/workflows/main.yml:63]: Does this run linting as well? Or just pytest?

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -0,0 +1,64 @@
+name: Tests
```
Comment (Saurav-D) [inline .github/workflows/main.yml:1]: Is this CI specifically for tests?

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+        id: cache
+        with:
+          path: .venv
+          key: venv-${{ steps.full-python-version.outputs.version }}-${{ hashFiles('**/poetry.lock') }}
+      - name: Ensure cache is healthy
+        if: steps.cache.outputs.cache-hit == 'true'
+        shell: bash
+        run: poetry run pip --version >/dev/null 2>&1 || rm -rf .venv
+      - name: Install dependencies
+        run: poetry install
```
Comment (Saurav-D) [inline .github/workflows/main.yml:61]: Can we add black here for code formatting as well?

Comment (adason) [reply .github/workflows/main.yml:61]: Yes. `pre-commit run --all-files` will run everything include flake8, black, isort,...

File: .github/workflows/main.yml (hunk 2 above)
Comment (adason) [reply .github/workflows/main.yml:1]: Yes. We will have a separate CI yaml that does other things such as push to dockuerhub and push to pypi.

**Comment by Saurav-D** [comment]
Looks good.

**Review by Saurav-D — APPROVED** [review]

**Pull request merged**